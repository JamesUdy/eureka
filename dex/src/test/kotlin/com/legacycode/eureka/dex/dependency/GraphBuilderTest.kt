package com.legacycode.eureka.dex.dependency

import com.legacycode.eureka.dex.test.TestApk
import org.approvaltests.Approvals
import org.junit.jupiter.api.BeforeEach
import org.junit.jupiter.api.Test

class GraphBuilderTest {
  private lateinit var parser: DependencyApkParser

  @BeforeEach
  fun setUp() {
    val testApk = TestApk("wikipedia.apk")
    parser = DependencyApkParser(testApk.file)
  }

  @Test
  fun `build a graphviz graph`() {
    // when
    val adjacencyList = parser.buildDependencyGraph()

    // then
    val dependencyGraph = adjacencyList.graph(DotGraphBuilder("wikipedia.apk (Flows)"))
    Approvals.verify(dependencyGraph)
  }

  @Test
  fun `build a ontology graph`() {
    // when
    val adjacencyList = parser.buildDependencyGraph()

    // then
    val dependencyGraph = adjacencyList.graph(OntologyGraphBuilder())
    Approvals.verify(dependencyGraph)
  }
}
