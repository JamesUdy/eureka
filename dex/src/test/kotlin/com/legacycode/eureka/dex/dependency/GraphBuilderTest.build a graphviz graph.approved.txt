digraph "wikipedia.apk (Flows)" {
  "androidx.lifecycle.ReportFragment" -> "androidx.core.app.ComponentActivity"
  "androidx.lifecycle.ReportFragment" -> "androidx.activity.ComponentActivity"
  "androidx.preference.EditTextPreferenceDialogFragmentCompat" -> "androidx.preference.PreferenceFragmentCompat"
  "androidx.preference.ListPreferenceDialogFragmentCompat" -> "androidx.preference.PreferenceFragmentCompat"
  "androidx.preference.MultiSelectListPreferenceDialogFragmentCompat" -> "androidx.preference.PreferenceFragmentCompat"
  "org.wikipedia.descriptions.DescriptionEditActivity" -> "org.wikipedia.commons.FilePageFragment"
  "org.wikipedia.suggestededits.SuggestedEditsImageTagEditActivity" -> "org.wikipedia.commons.FilePageFragment"
  "org.wikipedia.login.LoginActivity" -> "org.wikipedia.descriptions.DescriptionEditFragment"
  "org.wikipedia.descriptions.DescriptionEditSuccessActivity" -> "org.wikipedia.descriptions.DescriptionEditFragment"
  "org.wikipedia.descriptions.DescriptionEditTutorialFragment" -> "org.wikipedia.descriptions.DescriptionEditTutorialFragment$ItemFragment"
  "org.wikipedia.commons.FilePageActivity" -> "org.wikipedia.diff.ArticleEditDetailsFragment"
  "org.wikipedia.page.linkpreview.LinkPreviewDialog" -> "org.wikipedia.diff.ArticleEditDetailsFragment"
  "org.wikipedia.talk.TalkTopicsActivity" -> "org.wikipedia.diff.ArticleEditDetailsFragment"
  "org.wikipedia.watchlist.WatchlistExpiryDialog" -> "org.wikipedia.diff.ArticleEditDetailsFragment"
  "org.wikipedia.page.PageActivity" -> "org.wikipedia.diff.ArticleEditDetailsFragment"
  "org.wikipedia.edit.insertmedia.InsertMediaActivity" -> "org.wikipedia.edit.insertmedia.InsertMediaAdvancedSettingsFragment"
  "org.wikipedia.edit.insertmedia.InsertMediaImagePositionDialog" -> "org.wikipedia.edit.insertmedia.InsertMediaAdvancedSettingsFragment"
  "org.wikipedia.edit.insertmedia.InsertMediaImageTypeDialog" -> "org.wikipedia.edit.insertmedia.InsertMediaAdvancedSettingsFragment"
  "org.wikipedia.edit.insertmedia.InsertMediaImageSizeDialog" -> "org.wikipedia.edit.insertmedia.InsertMediaAdvancedSettingsFragment"
  "org.wikipedia.edit.insertmedia.InsertMediaActivity" -> "org.wikipedia.edit.insertmedia.InsertMediaSettingsFragment"
  "org.wikipedia.views.ImagePreviewDialog" -> "org.wikipedia.edit.insertmedia.InsertMediaSettingsFragment"
  "org.wikipedia.page.references.ReferenceDialog" -> "org.wikipedia.edit.preview.EditPreviewFragment"
  "org.wikipedia.edit.EditSectionActivity" -> "org.wikipedia.edit.preview.EditPreviewFragment"
  "org.wikipedia.edit.EditSectionActivity" -> "org.wikipedia.edit.summaries.EditSummaryFragment"
  "org.wikipedia.feed.configure.ConfigureActivity" -> "org.wikipedia.feed.FeedFragment"
  "org.wikipedia.settings.languages.WikipediaLanguagesActivity" -> "org.wikipedia.feed.FeedFragment"
  "org.wikipedia.feed.news.NewsActivity" -> "org.wikipedia.feed.news.NewsFragment"
  "org.wikipedia.views.CustomDatePicker" -> "org.wikipedia.feed.onthisday.OnThisDayFragment"
  "org.wikipedia.commons.FilePageActivity" -> "org.wikipedia.feed.suggestededits.SuggestedEditsCardItemFragment"
  "org.wikipedia.page.PageActivity" -> "org.wikipedia.feed.suggestededits.SuggestedEditsCardItemFragment"
  "org.wikipedia.gallery.GalleryActivity" -> "org.wikipedia.feed.suggestededits.SuggestedEditsCardItemFragment"
  "org.wikipedia.suggestededits.SuggestedEditsImageTagEditActivity" -> "org.wikipedia.feed.suggestededits.SuggestedEditsCardItemFragment"
  "org.wikipedia.descriptions.DescriptionEditActivity" -> "org.wikipedia.feed.suggestededits.SuggestedEditsCardItemFragment"
  "com.google.android.material.datepicker.MaterialDatePicker" -> "com.google.android.material.datepicker.MaterialCalendar"
  "com.google.android.material.datepicker.MaterialCalendar" -> "com.google.android.material.datepicker.MaterialDatePicker"
  "com.google.android.material.datepicker.MaterialTextInputPicker" -> "com.google.android.material.datepicker.MaterialDatePicker"
  "org.wikipedia.login.LoginActivity" -> "org.wikipedia.activity.BaseActivity"
  "org.wikipedia.page.linkpreview.LinkPreviewDialog" -> "org.wikipedia.categories.CategoryActivity"
  "org.wikipedia.page.PageActivity" -> "org.wikipedia.categories.CategoryActivity"
  "org.wikipedia.categories.CategoryDialog" -> "org.wikipedia.categories.CategoryActivity"
  "org.wikipedia.edit.preview.EditPreviewFragment" -> "org.wikipedia.edit.EditSectionActivity"
  "org.wikipedia.edit.summaries.EditSummaryFragment" -> "org.wikipedia.edit.EditSectionActivity"
  "org.wikipedia.login.LoginActivity" -> "org.wikipedia.edit.EditSectionActivity"
  "org.wikipedia.theme.ThemeChooserDialog" -> "org.wikipedia.edit.EditSectionActivity"
  "org.wikipedia.edit.insertmedia.InsertMediaAdvancedSettingsFragment" -> "org.wikipedia.edit.insertmedia.InsertMediaActivity"
  "org.wikipedia.edit.insertmedia.InsertMediaSettingsFragment" -> "org.wikipedia.edit.insertmedia.InsertMediaActivity"
  "org.wikipedia.commons.FilePageActivity" -> "org.wikipedia.edit.insertmedia.InsertMediaActivity"
  "org.wikipedia.commons.FilePageFragment" -> "org.wikipedia.commons.FilePageActivity"
  "org.wikipedia.descriptions.DescriptionEditFragment" -> "org.wikipedia.descriptions.DescriptionEditActivity"
  "org.wikipedia.page.linkpreview.LinkPreviewDialog" -> "org.wikipedia.descriptions.DescriptionEditActivity"
  "org.wikipedia.views.ImagePreviewDialog" -> "org.wikipedia.descriptions.DescriptionEditActivity"
  "org.wikipedia.descriptions.DescriptionEditTutorialActivity" -> "org.wikipedia.descriptions.DescriptionEditActivity"
  "org.wikipedia.page.PageActivity" -> "org.wikipedia.descriptions.DescriptionEditActivity"
  "org.wikipedia.descriptions.DescriptionEditTutorialFragment" -> "org.wikipedia.descriptions.DescriptionEditTutorialActivity"
  "org.wikipedia.diff.ArticleEditDetailsFragment" -> "org.wikipedia.diff.ArticleEditDetailsActivity"
  "org.wikipedia.edit.insertmedia.InsertMediaActivity" -> "org.wikipedia.edit.insertmedia.InsertMediaImagePositionDialog"
  "org.wikipedia.edit.insertmedia.InsertMediaActivity" -> "org.wikipedia.edit.insertmedia.InsertMediaImageSizeDialog"
  "org.wikipedia.edit.insertmedia.InsertMediaActivity" -> "org.wikipedia.edit.insertmedia.InsertMediaImageTypeDialog"
  "org.wikipedia.feed.configure.ConfigureFragment" -> "org.wikipedia.feed.configure.ConfigureActivity"
  "org.wikipedia.feed.news.NewsFragment" -> "org.wikipedia.feed.news.NewsActivity"
  "org.wikipedia.feed.onthisday.OnThisDayFragment" -> "org.wikipedia.feed.onthisday.OnThisDayActivity"
  "org.wikipedia.feed.topread.TopReadFragment" -> "org.wikipedia.feed.topread.TopReadArticlesActivity"
  "org.wikipedia.descriptions.DescriptionEditSuccessFragment" -> "org.wikipedia.descriptions.DescriptionEditSuccessActivity"
  "org.wikipedia.gallery.GalleryActivity" -> "org.wikipedia.gallery.GalleryItemFragment"
  "org.wikipedia.commons.FilePageActivity" -> "org.wikipedia.gallery.GalleryItemFragment"
  "org.wikipedia.suggestededits.SuggestedEditsTasksFragment" -> "org.wikipedia.main.MainFragment"
  "org.wikipedia.navtab.MenuNavTabDialog" -> "org.wikipedia.main.MainFragment"
  "org.wikipedia.feed.FeedFragment" -> "org.wikipedia.main.MainFragment"
  "org.wikipedia.history.HistoryFragment" -> "org.wikipedia.main.MainFragment"
  "org.wikipedia.page.PageActivity" -> "org.wikipedia.main.MainFragment"
  "org.wikipedia.page.tabs.TabActivity" -> "org.wikipedia.main.MainFragment"
  "org.wikipedia.notifications.NotificationActivity" -> "org.wikipedia.main.MainFragment"
  "org.wikipedia.readinglist.ReadingListsFragment" -> "org.wikipedia.main.MainFragment"
  "org.wikipedia.usercontrib.UserContribListActivity" -> "org.wikipedia.main.MainFragment"
  "org.wikipedia.random.RandomActivity" -> "org.wikipedia.main.MainFragment"
  "org.wikipedia.commons.FilePageActivity" -> "org.wikipedia.main.MainFragment"
  "org.wikipedia.readinglist.MoveToReadingListDialog" -> "org.wikipedia.main.MainFragment"
  "org.wikipedia.feed.news.NewsActivity" -> "org.wikipedia.main.MainFragment"
  "org.wikipedia.login.LoginActivity" -> "org.wikipedia.main.MainFragment"
  "org.wikipedia.search.SearchActivity" -> "org.wikipedia.main.MainFragment"
  "org.wikipedia.settings.SettingsActivity" -> "org.wikipedia.main.MainFragment"
  "org.wikipedia.talk.TalkTopicsActivity" -> "org.wikipedia.main.MainFragment"
  "org.wikipedia.watchlist.WatchlistActivity" -> "org.wikipedia.main.MainFragment"
  "org.wikipedia.page.PageActivity" -> "org.wikipedia.page.PageFragment"
  "org.wikipedia.categories.CategoryActivity" -> "org.wikipedia.page.PageFragment"
  "org.wikipedia.page.references.ReferenceDialog" -> "org.wikipedia.page.PageFragment"
  "org.wikipedia.page.edithistory.EditHistoryListActivity" -> "org.wikipedia.page.PageFragment"
  "org.wikipedia.watchlist.WatchlistExpiryDialog" -> "org.wikipedia.page.PageFragment"
  "org.wikipedia.gallery.GalleryActivity" -> "org.wikipedia.page.PageFragment"
  "org.wikipedia.talk.TalkTopicsActivity" -> "org.wikipedia.page.PageFragment"
  "org.wikipedia.login.LoginActivity" -> "org.wikipedia.page.PageFragment"
  "org.wikipedia.main.MainActivity" -> "org.wikipedia.page.PageFragment"
  "org.wikipedia.theme.ThemeChooserDialog" -> "org.wikipedia.page.PageFragment"
  "org.wikipedia.random.RandomItemFragment" -> "org.wikipedia.random.RandomFragment"
  "org.wikipedia.readinglist.MoveToReadingListDialog" -> "org.wikipedia.random.RandomFragment"
  "org.wikipedia.page.PageActivity" -> "org.wikipedia.random.RandomFragment"
  "org.wikipedia.random.RandomFragment" -> "org.wikipedia.random.RandomItemFragment"
  "org.wikipedia.main.MainActivity" -> "org.wikipedia.readinglist.ReadingListFragment"
  "org.wikipedia.readinglist.MoveToReadingListDialog" -> "org.wikipedia.readinglist.ReadingListFragment"
  "org.wikipedia.readinglist.ReadingListsFragment" -> "org.wikipedia.readinglist.ReadingListFragment"
  "org.wikipedia.readinglist.ReadingListActivity" -> "org.wikipedia.readinglist.ReadingListsFragment"
  "org.wikipedia.readinglist.MoveToReadingListDialog" -> "org.wikipedia.readinglist.ReadingListsFragment"
  "org.wikipedia.main.MainActivity" -> "org.wikipedia.readinglist.ReadingListsFragment"
  "org.wikipedia.search.RecentSearchesFragment" -> "org.wikipedia.search.SearchFragment"
  "org.wikipedia.search.SearchResultsFragment" -> "org.wikipedia.search.SearchFragment"
  "org.wikipedia.settings.languages.WikipediaLanguagesActivity" -> "org.wikipedia.search.SearchFragment"
  "org.wikipedia.page.PageActivity" -> "org.wikipedia.search.SearchFragment"
  "org.wikipedia.readinglist.MoveToReadingListDialog" -> "org.wikipedia.search.SearchFragment"
  "org.wikipedia.search.SearchFragment" -> "org.wikipedia.search.SearchResultsFragment"
  "org.wikipedia.suggestededits.SuggestedEditsCardsItemFragment" -> "org.wikipedia.suggestededits.SuggestedEditsCardsFragment"
  "org.wikipedia.suggestededits.SuggestedEditsImageTagsOnboardingActivity" -> "org.wikipedia.suggestededits.SuggestedEditsCardsFragment"
  "org.wikipedia.descriptions.DescriptionEditActivity" -> "org.wikipedia.suggestededits.SuggestedEditsCardsFragment"
  "org.wikipedia.suggestededits.SuggestedEditsCardsFragment" -> "org.wikipedia.suggestededits.SuggestedEditsItemFragment"
  "org.wikipedia.usercontrib.UserContribListActivity" -> "org.wikipedia.suggestededits.SuggestedEditsTasksFragment"
  "org.wikipedia.main.MainActivity" -> "org.wikipedia.suggestededits.SuggestedEditsTasksFragment"
  "org.wikipedia.suggestededits.SuggestionsActivity" -> "org.wikipedia.suggestededits.SuggestedEditsTasksFragment"
  "org.wikipedia.notifications.NotificationActivity" -> "org.wikipedia.watchlist.WatchlistFragment"
  "org.wikipedia.watchlist.WatchlistFilterActivity" -> "org.wikipedia.watchlist.WatchlistFragment"
  "org.wikipedia.diff.ArticleEditDetailsActivity" -> "org.wikipedia.watchlist.WatchlistFragment"
  "org.wikipedia.login.LoginActivity" -> "org.wikipedia.onboarding.InitialOnboardingFragment"
  "org.wikipedia.settings.languages.WikipediaLanguagesActivity" -> "org.wikipedia.onboarding.InitialOnboardingFragment"
  "org.wikipedia.suggestededits.SuggestedEditsCardsFragment" -> "org.wikipedia.suggestededits.SuggestedEditsCardsItemFragment"
  "org.wikipedia.commons.FilePageActivity" -> "org.wikipedia.suggestededits.SuggestedEditsImageTagsFragment"
  "org.wikipedia.suggestededits.SuggestedEditsImageTagDialog" -> "org.wikipedia.suggestededits.SuggestedEditsImageTagsFragment"
  "org.wikipedia.settings.DeveloperSettingsActivity" -> "org.wikipedia.settings.SettingsFragment"
  "org.wikipedia.gallery.GalleryItemFragment" -> "org.wikipedia.gallery.GalleryActivity"
  "org.wikipedia.page.PageActivity" -> "org.wikipedia.gallery.GalleryActivity"
  "org.wikipedia.commons.FilePageActivity" -> "org.wikipedia.gallery.GalleryActivity"
  "org.wikipedia.page.linkpreview.LinkPreviewDialog" -> "org.wikipedia.gallery.GalleryActivity"
  "org.wikipedia.descriptions.DescriptionEditActivity" -> "org.wikipedia.gallery.GalleryActivity"
  "org.wikipedia.suggestededits.SuggestedEditsImageTagEditActivity" -> "org.wikipedia.gallery.GalleryActivity"
  "org.wikipedia.createaccount.CreateAccountActivity" -> "org.wikipedia.login.LoginActivity"
  "org.wikipedia.createaccount.CreateAccountActivity" -> "org.wikipedia.login.ResetPasswordActivity"
  "org.wikipedia.notifications.NotificationFilterActivity" -> "org.wikipedia.notifications.NotificationActivity"
  "org.wikipedia.settings.NotificationSettingsActivity" -> "org.wikipedia.notifications.NotificationActivity"
  "org.wikipedia.page.PageFragment" -> "org.wikipedia.page.PageActivity"
  "org.wikipedia.readinglist.ReadingListActivity" -> "org.wikipedia.page.PageActivity"
  "org.wikipedia.commons.FilePageActivity" -> "org.wikipedia.page.PageActivity"
  "org.wikipedia.usercontrib.UserContribListActivity" -> "org.wikipedia.page.PageActivity"
  "org.wikipedia.talk.TalkTopicsActivity" -> "org.wikipedia.page.PageActivity"
  "org.wikipedia.search.SearchActivity" -> "org.wikipedia.page.PageActivity"
  "org.wikipedia.page.tabs.TabActivity" -> "org.wikipedia.page.PageActivity"
  "org.wikipedia.notifications.NotificationActivity" -> "org.wikipedia.page.PageActivity"
  "org.wikipedia.gallery.GalleryActivity" -> "org.wikipedia.page.PageActivity"
  "org.wikipedia.suggestededits.SuggestedEditsImageTagEditActivity" -> "org.wikipedia.page.PageActivity"
  "org.wikipedia.descriptions.DescriptionEditActivity" -> "org.wikipedia.page.PageActivity"
  "org.wikipedia.edit.EditSectionActivity" -> "org.wikipedia.page.PageActivity"
  "org.wikipedia.language.LangLinksActivity" -> "org.wikipedia.page.PageActivity"
  "org.wikipedia.page.linkpreview.LinkPreviewDialog" -> "org.wikipedia.page.PageActivity"
  "org.wikipedia.diff.ArticleEditDetailsActivity" -> "org.wikipedia.page.edithistory.EditHistoryListActivity"
  "org.wikipedia.main.MainActivity" -> "org.wikipedia.page.tabs.TabActivity"
  "org.wikipedia.notifications.NotificationActivity" -> "org.wikipedia.page.tabs.TabActivity"
  "org.wikipedia.page.PageActivity" -> "org.wikipedia.page.tabs.TabActivity"
  "org.wikipedia.suggestededits.SuggestedEditsImageTagsFragment" -> "org.wikipedia.suggestededits.SuggestedEditsImageTagEditActivity"
  "org.wikipedia.suggestededits.SuggestedEditsImageTagsOnboardingActivity" -> "org.wikipedia.suggestededits.SuggestedEditsImageTagEditActivity"
  "org.wikipedia.talk.TalkTopicsActivity" -> "org.wikipedia.talk.ArchivedTalkPagesActivity"
  "org.wikipedia.page.PageActivity" -> "org.wikipedia.talk.ArchivedTalkPagesActivity"
  "org.wikipedia.talk.TalkTopicsActivity" -> "org.wikipedia.talk.TalkReplyActivity"
  "org.wikipedia.page.PageActivity" -> "org.wikipedia.talk.TalkReplyActivity"
  "org.wikipedia.login.LoginActivity" -> "org.wikipedia.talk.TalkReplyActivity"
  "org.wikipedia.talk.TalkReplyActivity" -> "org.wikipedia.talk.TalkTopicActivity"
  "org.wikipedia.page.PageActivity" -> "org.wikipedia.talk.TalkTopicActivity"
  "org.wikipedia.edit.EditSectionActivity" -> "org.wikipedia.talk.TalkTopicActivity"
  "org.wikipedia.page.PageActivity" -> "org.wikipedia.talk.TalkTopicsActivity"
  "org.wikipedia.talk.TalkReplyActivity" -> "org.wikipedia.talk.TalkTopicsActivity"
  "org.wikipedia.notifications.NotificationActivity" -> "org.wikipedia.talk.TalkTopicsActivity"
  "org.wikipedia.watchlist.WatchlistExpiryDialog" -> "org.wikipedia.talk.TalkTopicsActivity"
  "org.wikipedia.talk.TalkTopicActivity" -> "org.wikipedia.talk.TalkTopicsActivity"
  "org.wikipedia.settings.languages.WikipediaLanguagesActivity" -> "org.wikipedia.talk.TalkTopicsActivity"
  "org.wikipedia.page.edithistory.EditHistoryListActivity" -> "org.wikipedia.talk.TalkTopicsActivity"
  "org.wikipedia.talk.ArchivedTalkPagesActivity" -> "org.wikipedia.talk.TalkTopicsActivity"
  "org.wikipedia.page.PageActivity" -> "org.wikipedia.main.MainActivity"
  "org.wikipedia.main.MainFragment" -> "org.wikipedia.main.MainActivity"
  "org.wikipedia.onboarding.InitialOnboardingActivity" -> "org.wikipedia.main.MainActivity"
  "org.wikipedia.onboarding.InitialOnboardingFragment" -> "org.wikipedia.onboarding.InitialOnboardingActivity"
  "org.wikipedia.page.customize.CustomizeToolbarFragment" -> "org.wikipedia.page.customize.CustomizeToolbarActivity"
  "org.wikipedia.gallery.GalleryActivity" -> "org.wikipedia.page.linkpreview.LinkPreviewDialog"
  "org.wikipedia.random.RandomFragment" -> "org.wikipedia.random.RandomActivity"
  "org.wikipedia.readinglist.ReadingListActivity" -> "org.wikipedia.readinglist.AddToReadingListDialog"
  "org.wikipedia.readinglist.ReadingListFragment" -> "org.wikipedia.readinglist.ReadingListActivity"
  "org.wikipedia.search.SearchFragment" -> "org.wikipedia.search.SearchActivity"
  "org.wikipedia.settings.DeveloperSettingsFragment" -> "org.wikipedia.settings.DeveloperSettingsActivity"
  "org.wikipedia.settings.NotificationSettingsFragment" -> "org.wikipedia.settings.NotificationSettingsActivity"
  "org.wikipedia.settings.SettingsFragment" -> "org.wikipedia.settings.SettingsActivity"
  "org.wikipedia.settings.languages.WikipediaLanguagesFragment" -> "org.wikipedia.settings.languages.WikipediaLanguagesActivity"
  "org.wikipedia.suggestededits.SuggestedEditsCardsFragment" -> "org.wikipedia.suggestededits.SuggestionsActivity"
  "org.wikipedia.theme.ThemeChooserDialog" -> "org.wikipedia.theme.ThemeFittingRoomActivity"
  "org.wikipedia.theme.ThemeFittingRoomFragment" -> "org.wikipedia.theme.ThemeFittingRoomActivity"
  "org.wikipedia.watchlist.WatchlistFragment" -> "org.wikipedia.watchlist.WatchlistActivity"
}
